tests/communication/test_decorator.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from concurrent.futures import ThreadPoolExecutor, as_completed
+import os
+import subprocess
+import sys
+import threading
+import unittest
+
+from maro.communication import Proxy, SessionMessage, dist
+from utils import get_random_port, proxy_generator
+
+
+def handler_function(that, proxy, message):
+    replied_payload = {"counter": message.payload["counter"] + 1}
+    proxy.reply(message, payload=replied_payload)
+    sys.exit(0)
+
+
+def lunch_receiver(proxy, handler_dict):
+    @dist(proxy, handler_dict)
```

Comment:
move receiver class out of the lunch_receiver

Reply:
can not move receiver class outside, cause the dist decorator need the proxy and handler_dict for receiver class.